include/ultra64/asm.h

```diff
@@ -0,0 +1,74 @@
+
+#define LEAF(x)                 \
+    .balign 4                  ;\
+    .globl x                   ;\
```

Comment:
`IPL_SYMBOL` uses `.global`, may want to decide on one or the other unless there's some compatibility reason?

Reply:
I'll also answer more generally about `asm.h` here.

`asm.h` is an sdk header (https://github.com/n64decomp/libreultra/blob/master/include/2.0I/sys/asm.h) which along with `regdef.h` (same sys directory) are designed to try to make assembly more portable between assemblers (in particular, libultra was assembled by both IDO's assembler and GAS and while they're very similar they're not quite the same) and MIPS ABIs (o32, n32, 64) by hiding a number of assembler-specific syntax behind preprocessor macros. I felt like BEGIN and such were so close to these that we might as well use the original, and I wanted to vacate `macro.inc` to only contain `glabel` again. 

The macros are lifted from `asm.h` with some minor tweaks like additional `.balign` and `.type` directives. I prefer to have more complete sdk headers even if some of their contents is not used, since others may find new use for them in modding.

`LEAF` as a name stems from "leaf function", one that doesn't call other functions, although this is something of a misnomer since you can use these just fine with non-leaf functions. I unfortunately can't refer you to any method to lodge a complaint to SGI about the choice of name.

I think `.globl` is supported by both IDO and GAS, while `.global` is only supported by the latter, but I haven't checked this since I'd need to set up `as0` in QEMU since there's no working recomp for it...
I didn't use .globl in ipl3 since I'm not sure if it's really portable anyway. Again this would be nice to check, but same reason above.